### PR TITLE
test(unit): add InvoiceRepository and UserRepository unit tests

### LINKED ISSUE #491: test: add unit tests for invoice and user repositories
## Summary

The invoice and user repositories have ~21% statement coverage each with 0% branch coverage. These data-access layer components should be tested with a stubbed DB client.

## Files to cover

- `src/repositories/invoice-repository.ts` (21% statements, 0% branches)
- `src/repositories/user-repository.ts` (21% statements, 0% branches)

## Out of scope

- `src/repositories/event-repository.ts` — already at 78%/74%

## Acceptance criteria

- [ ] Each repository has a corresponding `.spec.ts` under `test/unit/repositories/`
- [ ] All public methods covered (CRUD, edge cases)
- [ ] DB client stubbed via sinon — no real DB connections
- [ ] `npm run cover:unit` passes without regressions

### PR DESCRIPTION
**Pull request by CKodidela**
## Description
  Add unit tests for `InvoiceRepository` and `UserRepository` — two data-access layer components that had ~21% statement
   coverage and 0% branch coverage. All public methods are now covered using a sinon-stubbed DB client with no real
  database connections.

  ## Related Issue
  Closes #491 

  ## Motivation and Context
  Both repositories lacked meaningful test coverage, making regressions in DB query logic hard to catch. These tests
  cover all public CRUD methods and edge cases (missing rows, error propagation, default values, query structure).

  ## How Has This Been Tested?
  - 20 new tests for `InvoiceRepository` under `test/unit/repositories/invoice-repository.spec.ts`
  - 21 new tests for `UserRepository` under `test/unit/repositories/user-repository.spec.ts`
  - Full test suite (1002 tests) passes with no regressions
  - DB client stubbed via sinon throughout — no real DB connections used

  Coverage after this PR:

  | File | Statements | Branches |
  |---|---|---|
  | `invoice-repository.ts` | 21% → 88% | 0% → 44% |
  | `user-repository.ts` | 21% → 93% | 0% → 67% |

  ## Types of changes
  - [x] Non-functional change (docs, style, minor refactor)

  ## Checklist:
  - [x] My code follows the code style of this project.
  - [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
  - [x] I have read the **CONTRIBUTING** document.
  - [x] I have added tests to cover my code changes.
  - [x] I added a changeset, or this is docs-only and I added an empty changeset.
  - [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/repositories/invoice-repository.spec.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      const { id: _id, ...invoiceWithoutId } = testInvoice
+      const sql = repository.upsert(invoiceWithoutId as Invoice).toString()
+
+      expect(sql).to.include('insert into "invoices"')
+    })
+
+    it('encodes pubkey as hex buffer in SQL', () => {
+      const sql = repository.upsert(testInvoice).toString()
+
+      expect(sql).to.include(pubkeyHex)
```
[the remark this answers is not in the document]
Comment (CKodidela) [reply test/unit/repositories/invoice-repository.spec.ts]: updated the assertion to check for the Knex/PG buffer

File: test/unit/repositories/user-repository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+      expect(sql).to.include('insert into "users"')
+      expect(sql).to.include('"is_vanished"')
+      expect(sql).to.include('on conflict')
+    })
+
+    it('encodes pubkey as hex buffer in SQL', () => {
+      const sql = repository.setVanished(pubkeyHex, false).toString()
+
+      expect(sql).to.include(pubkeyHex)
```
[the remark this answers is not in the document]
Comment (CKodidela) [reply test/unit/repositories/user-repository.spec.ts]: updated

File: test/unit/repositories/invoice-repository.spec.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+      expect(sql).to.include('specific-id')
+    })
+
+    it('uses a generated UUID when no id is provided', () => {
+      const { id: _id, ...invoiceWithoutId } = testInvoice
+      const sql = repository.upsert(invoiceWithoutId as Invoice).toString()
+
+      expect(sql).to.include('insert into "invoices"')
+    })
```
[the remark this answers is not in the document]
Comment (CKodidela) [reply test/unit/repositories/invoice-repository.spec.ts:251]: this suggestion is valid , updated to match and pushed the fix
